src/plugins/console/public/application/containers/remote/remote_console.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import {
+  EuiPortal,
+  EuiScreenReaderOnly,
+  EuiThemeProvider,
+  EuiWindowEvent,
+  keys,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { RemoteConsoleProps, RemoteConsoleDependencies } from '../../../types/remote_console';
+
+import { ConsoleHeader } from './console_header';
+import { ConsoleWrapper } from './console_wrapper';
+
+import './_index.scss';
+
+const landmarkHeading = i18n.translate('console.remoteConsole.landmarkHeading', {
+  defaultMessage: 'Developer console',
+});
+
+export const RemoteConsole = ({
+  headerRightSideItem,
+  size = 'm',
+  core,
+  usageCollection,
+}: RemoteConsoleProps & RemoteConsoleDependencies) => {
+  const [isConsoleOpen, setIsConsoleOpen] = useState<boolean>(false);
+  const toggleConsole = () => setIsConsoleOpen(!isConsoleOpen);
+
+  const onKeyDown = (event: any) => {
+    if (event.key === keys.ESCAPE) {
+      event.preventDefault();
+      event.stopPropagation();
+      setIsConsoleOpen(false);
+    }
+  };
+
+  const classes = classNames('remoteConsole', {
+    'remoteConsole-isOpen': isConsoleOpen,
+    'remoteConsole--large': size === 'l',
+    'remoteConsole--medium': size === 'm',
+    'remoteConsole--small': size === 's',
+    'remoteConsole--fixed': true,
+    'remoteConsole--showOnMobile': false,
+  });
+
+  return (
+    <EuiPortal>
+      <section aria-label={landmarkHeading} className={classes}>
+        <EuiScreenReaderOnly>
+          <h2>{landmarkHeading}</h2>
+        </EuiScreenReaderOnly>
+        <EuiThemeProvider colorMode={'dark'} wrapperProps={{ cloneElement: true }}>
+          <div className="remoteConsole__controls">
+            <ConsoleHeader
+              isConsoleOpen={isConsoleOpen}
+              rightSideItem={headerRightSideItem}
+              onClick={toggleConsole}
+            />
+          </div>
+        </EuiThemeProvider>
+        {isConsoleOpen ? (
+          <div className="remoteConsole__content">
+            <EuiWindowEvent event="keydown" handler={onKeyDown} />
```

Comment:
The behavior of this currently isn't ideal if your focused in the console code editor as you have to press `Esc` twice to close the flyout. The first press is captured by the console.

I initially added a focus trap to this as well but given the nature of this remote console I have removed it for now. But I am curious of others thoughts on if we should include it so that a click outside the dev console will close it.

Reply:
Clicking outside the console to close it makes a lot of sense to me.